extensions/htlc/besu-eth/src/main/typescript/web-services/get-status-endpoint.ts

```diff
@@ -0,0 +1,86 @@
+import { Express, Request, Response } from "express";
+import {
+  Checks,
+  Logger,
+  LoggerProvider,
+  LogLevelDesc,
+} from "@hyperledger/cactus-common";
+import {
+  IExpressRequestHandler,
+  IWebServiceEndpoint,
+} from "@hyperledger/cactus-core-api";
+import { registerWebServiceEndpoint } from "@hyperledger/cactus-core";
+import { Web3SigningCredential } from "@hyperledger/cactus-plugin-ledger-connector-besu";
+import OAS from "../../json/openapi.json";
+import { PluginHtlcEthBesu } from "../plugin-htlc-eth-besu";
+export interface IGetStatusEndpointOptions {
+  logLevel?: LogLevelDesc;
+  plugin: PluginHtlcEthBesu;
+}
+export class GetStatusEndpoint implements IWebServiceEndpoint {
+  public static readonly CLASS_NAME = "GetStatusEndpoint";
+  private readonly log: Logger;
+
+  public get className(): string {
+    return GetStatusEndpoint.CLASS_NAME;
+  }
+
+  constructor(public readonly options: IGetStatusEndpointOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(options, `${fnTag} arg options`);
+    const level = this.options.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+  }
+  public getOASPath() {
+    return OAS.paths[
+      "/api/v1/plugins/@hyperledger/cactus-plugin-htlc-eth-besu/get-status"
+    ];
+  }
+  public getVerbLowerCase(): string {
+    const apiPath = this.getOASPath();
+    return apiPath.get["x-hyperledger-cactus"].http.verbLowerCase;
+  }
+  public getPath(): string {
+    const apiPath = this.getOASPath();
+    return apiPath.get["x-hyperledger-cactus"].http.path;
+  }
+  public getOperationId(): string {
+    return this.getOASPath().get.operationId;
+  }
+
+  public registerExpress(expressApp: Express): IWebServiceEndpoint {
+    registerWebServiceEndpoint(expressApp, this);
+    return this;
+  }
+
+  public getExpressRequestHandler(): IExpressRequestHandler {
+    return this.handleRequest.bind(this);
+  }
+
+  public async handleRequest(req: Request, res: Response): Promise<void> {
+    const fnTag = "GetStatusEndpoint#handleRequest()";
+    this.log.debug(`GET ${this.getPath()}`);
+    try {
+      const query = req.query["ids"]?.toString();
+      const ids = query?.split(",");
+      const contractAddress = req.query["contractAddress"];
+      const connectorId = req.query["connectorId"];
+      const web3SigningCredential = req.query["web3SigningCredential"];
+
+      const { callOutput } = await this.options.plugin.getStatus(
+        ids as string[],
+        contractAddress as string,
+        connectorId as string,
+        (web3SigningCredential as unknown) as Web3SigningCredential,
+      );
+      res.send(callOutput);
+    } catch (ex) {
+      this.log.error(`${fnTag} failed to serve request`, ex);
+      res.status(400).json({
```

Comment:
@AzaharaC This has to be a code 500 because if an exception was caught in this catch block that means that we validated in the try block that the request was good and decided to proceed with it.
The 400 status means that something was bad with the request which we have to determine in the try block and then have some if condition in there to conditionally either return with the happy path (HTTP 200) or with the bad request 400 if we detect an issue.